include/ur_api.h

```diff
@@ -2116,19 +2118,39 @@ urSamplerCreateWithNativeHandle(
 /// @brief USM memory property flags
 typedef uint32_t ur_usm_flags_t;
 typedef enum ur_usm_flag_t {
-    UR_USM_FLAG_BIAS_CACHED = UR_BIT(0),              ///< Allocation should be cached
-    UR_USM_FLAG_BIAS_UNCACHED = UR_BIT(1),            ///< Allocation should not be cached
-    UR_USM_FLAG_WRITE_COMBINED = UR_BIT(2),           ///< Memory should be allocated write-combined (WC)
-    UR_USM_FLAG_INITIAL_PLACEMENT_DEVICE = UR_BIT(3), ///< Optimize shared allocation for first access on the device
-    UR_USM_FLAG_INITIAL_PLACEMENT_HOST = UR_BIT(4),   ///< Optimize shared allocation for first access on the host
-    UR_USM_FLAG_DEVICE_READ_ONLY = UR_BIT(5),         ///< Memory is only possibly modified from the host, but read-only in all
```

Comment:
@veselypeta : UR_USM_FLAG_DEVICE_READ_ONLY was moved to UR_USM_DEVICE_MEM_FLAG_DEVICE_READ_ONLY to ur_usm_device_mem_flags_t, which is in ur_usm_device_desc_t, but ur_usm_device_desc_t is not passed to urUSMDeviceAlloc, we are still passing there ur_usm_desc_t. Same goes for ur_usm_host_desc_t and urUSMHostAlloc.

is that a miss or I'm missing something?

Reply:
or is that ur_usm_device_desc_t and ur_usm_host_desc_t are to be passed as pNext of ur_usm_desc_t?

Reply:
Yes the idea is to add the actual descriptor with the `pNext` pointer. I included an example for the shared alloc variant [here](https://github.com/oneapi-src/unified-runtime/blob/main/test/conformance/usm/urUSMSharedAlloc.cpp#L42)